spring-integration-mqtt/src/main/java/org/springframework/integration/mqtt/inbound/Mqttv5PahoMessageDrivenChannelAdapter.java

```diff
@@ -309,38 +331,54 @@ public void deliveryComplete(IMqttToken token) {
 
 	@Override
 	public void connectComplete(boolean reconnect, String serverURI) {
-		if (!reconnect) {
-			ApplicationEventPublisher applicationEventPublisher = getApplicationEventPublisher();
-			String[] topics = getTopic();
-			this.topicLock.lock();
-			try {
-				if (topics.length > 0) {
-					int[] requestedQos = getQos();
-					this.mqttClient.subscribe(topics, requestedQos).waitForCompletion(getCompletionTimeout());
-					String message = "Connected and subscribed to " + Arrays.toString(topics);
-					logger.debug(message);
-					if (applicationEventPublisher != null) {
-						applicationEventPublisher.publishEvent(new MqttSubscribedEvent(this, message));
-					}
-				}
-			}
-			catch (MqttException ex) {
-				if (applicationEventPublisher != null) {
-					applicationEventPublisher.publishEvent(new MqttConnectionFailedEvent(this, ex));
-				}
-				logger.error(ex, () -> "Error subscribing to " + Arrays.toString(topics));
-			}
-			finally {
-				this.topicLock.unlock();
-			}
+		if (reconnect) {
+			return;
```

Comment:
Nit-pick: this is strange construction. 
Won't this one look better:
```
if (!reconnect) {
	subscribeToAll();
}
```
?

Reply:
Agree. This was two-phase refactoring.
First I changed
```
if (!reconnect) {
  <long-long-code>
}
to 
if (reconnect) {
  return;
}
```
but then I extracted those long-long code to a method because I needed to reuse it 🙂